diff --git a/px4_ros2_cpp/src/components/mode_executor.cpp b/px4_ros2_cpp/src/components/mode_executor.cpp
index 78ec912..9879d19 100644
--- a/px4_ros2_cpp/src/components/mode_executor.cpp
+++ b/px4_ros2_cpp/src/components/mode_executor.cpp
@@ -396,7 +396,7 @@ ModeExecutorBase::ScheduledMode::ScheduledMode(rclcpp::Node& node,
   _mode_completed_sub = node.create_subscription<px4_msgs::msg::ModeCompleted>(
       topic_namespace_prefix + "fmu/out/mode_completed" +
           px4_ros2::getMessageNameVersion<px4_msgs::msg::ModeCompleted>(),
-      rclcpp::QoS(1).best_effort(), [this, &node](px4_msgs::msg::ModeCompleted::UniquePtr msg) {
+      rclcpp::QoS(1).reliable(), [this, &node](px4_msgs::msg::ModeCompleted::UniquePtr msg) {
         if (active() && msg->nav_state == static_cast<uint8_t>(_mode_id)) {
           RCLCPP_DEBUG(node.get_logger(), "Got matching ModeCompleted message, result: %i",
                        msg->result);
